lib/internal/modules/esm/package_config.js

```diff
@@ -53,48 +46,19 @@ function getPackageConfig(path, specifier, base) {
       exports: undefined,
       imports: undefined,
     };
-    packageJSONCache.set(path, packageConfig);
-    return packageConfig;
-  }
-
-  let packageJSON;
-  try {
-    packageJSON = JSONParse(source);
-  } catch (error) {
-    throw new ERR_INVALID_PACKAGE_CONFIG(
-      path,
-      (base ? `"${specifier}" from ` : '') + fileURLToPath(base || specifier),
-      error.message,
-    );
+    packageJSONCache.set(path, json);
+    return json;
   }
 
-  let { imports, main, name, type } = filterOwnProperties(packageJSON, ['imports', 'main', 'name', 'type']);
-  const exports = ObjectPrototypeHasOwnProperty(packageJSON, 'exports') ? packageJSON.exports : undefined;
-  if (typeof imports !== 'object' || imports === null) {
-    imports = undefined;
-  }
-  if (typeof main !== 'string') {
-    main = undefined;
-  }
-  if (typeof name !== 'string') {
-    name = undefined;
-  }
   // Ignore unknown types for forwards compatibility
-  if (type !== 'module' && type !== 'commonjs') {
-    type = 'none';
+  if (packageJSON.type !== 'module' && packageJSON.type !== 'commonjs') {
+    packageJSON.type = 'none';
   }
 
-  const packageConfig = {
-    pjsonPath: path,
-    exists: true,
-    main,
-    name,
-    type,
-    exports,
-    imports,
-  };
-  packageJSONCache.set(path, packageConfig);
-  return packageConfig;
+  packageJSON.pjsonPath = path;
+  packageJSON.exists = true;
+  packageJSONCache.set(path, packageJSON);
+  return packageJSON;
```

Comment:
Is there a particular reason to change how this was being done? If not, I think it would be better to not change these lines.

Reply:
Not particularly. My main assumption was that we dont need to create a new object and we can use the existing object returned from the C++, but as I look at it now, I don't see any benefit of this approach.